packages/drawer/src/Drawer.tsx

```diff
@@ -0,0 +1,149 @@
+import React, {
+  useRef, useContext, useState, useCallback,
+} from 'react';
+import { withTheme, ThemeContext, ThemeProvider } from 'styled-components';
+import { TransitionGroup, Transition } from 'react-transition-group';
+
+import { IThemeNamespace } from '@xcritical/theme';
+import Portal from '@xcritical/portal';
+import Blanket from '@xcritical/blanket';
+
+import { IDrawerProps, DrawerTheme } from './interfaces';
+import {
+  Wrapper,
+  Content,
+  Fade,
+  Slide,
+  SeparatorWrapper,
+  Separator,
+  AntiSelect,
+  IconWrapper,
+} from './styled';
+import { ArrowLeft, ArrowRight } from './Icons';
+
+
+export const PureDrawer: React.FC<IDrawerProps> = React.memo<IDrawerProps>(({
+  children,
+  appearance = 'default',
+  baseAppearance = 'default',
+  isOpen = false,
+  onOutsideClick,
+  theme,
+  minWidth = 30,
+  maxWidth = 1000,
+  isRTL = false,
+  isMovable = false,
+  withCloseButton = false,
+}: IDrawerProps) => {
+  const themeContext = useContext<IThemeNamespace<DrawerTheme>>(ThemeContext);
+
+  const clickX = useRef(0);
+  const drawerRef = useRef<HTMLElement>(null);
+
+  const [width, setWidth] = useState(maxWidth);
+  const [antiSelectLayer, setAntiSelectLayer] = useState(false);
+
+
+  const handleMouseMove = useCallback(
+    (e) => {
+      if (!isMovable) return;
+
+      setAntiSelectLayer(true);
+      document.body.addEventListener('mouseup', () => {
+        document.body.removeEventListener('mousemove', handleMouseMove);
+        setAntiSelectLayer(false);
+      });
+
+      const { clientX: currentX } = e;
+      const newWidth = isRTL
+        ? width - (currentX - clickX.current)
+        : width + (currentX - clickX.current);
+
+      if (newWidth >= maxWidth) return;
+
+      if (newWidth <= 0) {
+        document.body.removeEventListener('mousemove', handleMouseMove);
+        setWidth(minWidth);
+      } else if (newWidth <= minWidth) {
+        setWidth(minWidth);
+      } else {
+        setWidth(newWidth);
+      }
+    },
+    [isMovable, isRTL, maxWidth, minWidth, width],
+  );
+
+  const handleMouseDown = useCallback(
+    (e) => {
+      clickX.current = e.clientX;
+      document.body.addEventListener('mousemove', handleMouseMove);
+    },
+    [handleMouseMove],
+  );
+
+  const handleRemoveMouseMove = useCallback(() => {
```

Comment:
handleRemoveMouseMove -> handleMouseUp

Reply:
No, handleMouseUp say that we add event listener on 'mouseup' event, but in this method we only remove 'mousemove' event listener.